contributing/metadata.md

```diff
@@ -0,0 +1,166 @@
+
+The completeness is checked by comparing the registry to the metadata coverage report, but excluding
+
+1. Types that aren't supported in the metadata API
+2. Types in the [nonSupportedTypes file](../src/registry/nonSupportedTypes.ts) (think of it as a registry-ignore file). You can ignore the types themselves, or the feature/settings they depend on. Be sure to explain why you're choosing to ignore that type.
+
+## Manual Testing
+
+Want to make sure your types are working as expected?
+
+1. Create a new project with `sfdx force:project:create -n registryTest`
+1. Create a scratch org `sfdx force:org:create`
+1. Open the org and create your types.
+1. Run `sfdx force:source:beta:status` and verify the remote add.
+1. `sfdx force:source:beta:pull` to pull the metadata and examine what is retrieved
+1. Run `sfdx force:source:beta:status` and verify the changes were retrieved and no longer appear.
+1. Delete the org `sfdx force:org:delete --noprompt`
+1. Create a new scratch org. `sfdx force:org:create`
+1. Push the source `sfdx force:source:beta:push`
+1. Convert the source to mdapi format `sfdx force:source:convert -d mdapiOut`
+1. look in the resulting `metadataPackage_` and `package.xml` to see that it looks as expected
+1. deploy it to the org using `sfdx force:mdapi:deploy --deploydir mdapiOut --wait 30` and verify that it succeeds
+1. delete the source directory `rm -rf force-app/main/default/*`
+1. Create a new scratch org and convert the source back
+1. Convert back from mdapi to source format `sfdx force:mdapi:convert -r mdapiOut -d force-app`
+1. `sfdx force:source:beta:push`
+
+### Caveats
+
+Only source:push and source:pull support source tracking so the target types must be MDAPI addressable on the server. If they aren’t then special code is needed to support source tracking for these components. See the document [Metadata API Types: End to End, Cradle to Grave](https://docs.google.com/document/d/13jJLbM_ztS0tOOV8bQzNAnEFbL5MY4rNLa18oDV2SCs/edit?usp=sharing) (Salesforce internal only) for more details.
+
+## Unit Testing
+
+SDR Unit tests are abstract (mockRegistry tries to represent possible patterns). If your type uses a combination of properties that don't match an existing type in the registry, there's a chance the behavior isn't covered.
+
+Reach out to the CLI team for help with unit tests.
+
+[metadataResolverRealRegistry.ts](../test/resolve/metadataResolverRealRegistry.test.ts) is an example of unit testing one behavior (resolving from source files) of a real metadata type.
+
+## Integration Testing
+
+If you're doing anything complex (you've used any of the following properties `strategies`, `folderType`, `inFolder=true`, `ignoreParsedFullName`, `folderContentType`, `ignoreParentName`) then you'll wanted to add some NUTs that verify the behavior or your types using real orgs and to prevent SDR changes from causing regressions on your types.
```

Comment:
```suggestion
If you're doing anything complex (you've used any of the following properties `strategies`, `folderType`, `inFolder=true`, `ignoreParsedFullName`, `folderContentType`, `ignoreParentName`), you'll want to add some NUTs that verify the behavior or your types using real orgs and to prevent SDR changes from causing regressions on your types.
```